src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -476,6 +476,9 @@ private void updateScore(Result result, Set<ProgrammingExerciseTestCase> success
         }
         else {
             double weightSum = allTests.stream().filter(testCase -> !testCase.isInvisible()).mapToDouble(ProgrammingExerciseTestCase::getWeight).sum();
+            if (weightSum == 0) {
```

Comment:
afaik checking for equality via `==` for floating-point numbers is not a good idea. not sure if this is different in modern JDKs, but usually I would use `<= 0.001` or some other small epsilon to check if `weightSum` is very very close to zero. Maybe check this again: https://belief-driven-design.com/equality-and-comparison-in-java-a5e0f05b808/

Reply:
Puh I'm not too sure here. Does anyone else have experience with this?

Reply:
@lschlesinger is right. Typically, you would use something `if (Math.abs(weightSum - 0) < 1E-8)`

Reply:
Okay great, I'll add it in the code